test/modules/shadowing/unexpected-shadowing-warning/submodule-in-used-vs-toplevel-1.chpl

```diff
@@ -0,0 +1,17 @@
+module M {
+  module N {
+    var x = 100;
+  }
+}
+
+module N {
+  var x = 42;
+}
+
+module Application {
+  use N only;
+  use M;
+  proc main() {
+    writeln(N.x);
```

Comment:
This test case seems like a slight problem to me (but I don't think you changed its behavior).  `use N only;` in this test is basically useless (though the solution of writing `use M except N;` is probably reasonable and not burdensome enough to change)

Reply:
So it seems like getting warning in this case is a good thing? Was there some action you suggest here or was this more just your thoughts on the situation in this test?

Reply:
The warning is definitely useful.  I think I'm more displeased that we can make a prior use statement useless and would maybe want it to be an error in this situation, but that probably complicates the implementation and the rules too much.

Reply:
Ah, thanks for clarifying. Feel free to make an issue to advocate for this case being an error, if you feel strongly enough about it. Another possibility is that we could have this warning or a new warning say something about the order of the `use` statements being unintuitive. But, that too, would probably be best proposed and discussed on a different issue, because it's not completely obvious to me what logic we would want, and if it'd be a separate warning or an adjustment to this one.